crates/ra_ide/src/completion/completion_item.rs

```diff
@@ -333,12 +333,27 @@ impl<'a> Into<CompletionItem> for Builder {
 /// Represents an in-progress set of completions being built.
 #[derive(Debug, Default)]
 pub(crate) struct Completions {
+    existing: HashSet<LabelAndKind>,
     buf: Vec<CompletionItem>,
 }
 
+#[derive(Debug, Hash, PartialEq, Eq)]
+struct LabelAndKind {
+    label: String,
+    kind: Option<CompletionItemKind>,
+}
+impl LabelAndKind {
+    fn new(label: String, kind: Option<CompletionItemKind>) -> Self {
+        Self { label, kind }
+    }
+}
```

Comment:
I don't think there is any need for a struct like this. You can just use a tuple.